client/network/src/discovery.rs

```diff
@@ -539,6 +541,9 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 							for (addr, _error) in errors {
 								list.retain(|a| a != addr);
 							}
+							if list.is_empty() {
+								self.ephemeral_addresses.remove(&peer_id);
+							}
```

Comment:
if let Entry::Occupied(entry) = self.ephemeral_addresses.entry(peer_id) {
 							for (addr, _error) in errors {
								entry.get_mut().retain(|a| a != addr);
							}
							if entry.get().is_empty() {
								entry.remove();
							}
}

Reply:
This should replace the entire `if let Some` block